internal/telemetry/message.go

```diff
@@ -51,48 +73,67 @@ const (
-	LanguageVersion string   `json:"language_version"`
-	RuntimeName     string   `json:"runtime_name,omitempty"`
-	RuntimeVersion  string   `json:"runtime_version,omitempty"`
-	RuntimePatches  string   `json:"runtime_patches,omitempty"`
-	Products        Products `json:"products,omitempty"`
-}
-
-// Products specifies information about available products.
-type Products struct {
-	AppSec   ProductDetails `json:"appsec,omitempty"`
-	Profiler ProductDetails `json:"profiler,omitempty"`
-}
-
-// ProductDetails specifies details about a product.
-type ProductDetails struct {
-	Version string `json:"version"`
+	ServiceName     string `json:"service_name"`
+	Env             string `json:"env"`
+	ServiceVersion  string `json:"service_version"`
+	TracerVersion   string `json:"tracer_version"`
+	LanguageName    string `json:"language_name"`
+	LanguageVersion string `json:"language_version"`
+	RuntimeName     string `json:"runtime_name"`
+	RuntimeVersion  string `json:"runtime_version"`
+	RuntimePatches  string `json:"runtime_patches,omitempty"`
 }
 
 // Host is identifying information about the host on which the app
 // is running
 type Host struct {
-	ContainerID string `json:"container_id,omitempty"`
-	Hostname    string `json:"hostname,omitempty"`
-	OS          string `json:"os,omitempty"`
-	OSVersion   string `json:"os_version,omitempty"`
+	Hostname  string `json:"hostname"`
+	OS        string `json:"os"`
+	OSVersion string `json:"os_version,omitempty"`
 	// TODO: Do we care about the kernel stuff? internal/osinfo gets most of
 	// this information in OSName/OSVersion
-	KernelName    string `json:"kernel_name,omitempty"`
-	KernelRelease string `json:"kernel_release,omitempty"`
-	KernelVersion string `json:"kernel_version,omitempty"`
+	Architecture  string `json:"architecture"`
+	KernelName    string `json:"kernel_name"`
+	KernelRelease string `json:"kernel_release"`
+	KernelVersion string `json:"kernel_version"`
 }
 
 // AppStarted corresponds to the "app-started" request type
 type AppStarted struct {
-	Integrations  []Integration   `json:"integrations"`
-	Dependencies  []Dependency    `json:"dependencies"`
-	Configuration []Configuration `json:"configuration"`
+	Configuration     []Configuration     `json:"configuration,omitempty"`
+	Products          Products            `json:"products,omitempty"`
+	AdditionalPayload []AdditionalPayload `json:"additional_payload,omitempty"`
+	Error             Error               `json:"error,omitempty"`
+}
+
+// ConfigurationChange corresponds to the `AppClientConfigurationChange` event
+// that contains information about configuration changes since the app-started event
+type ConfigurationChange struct {
+	Configuration []Configuration `json:"conf_key_values"`
+	RemoteConfig  RemoteConfig    `json:"remote_config"`
+}
+
+// Configuration is a library-specific configuration value
+type Configuration struct {
+	Name string `json:"name"`
+	// Value should have a type that can be marshaled to JSON
```

Comment:
Is this comment saying that Value is a type that's able to be marshaled to JSON, or is this a TODO? Wondering if it should be removed...

Reply:
It was a TODO to enforce `Value` is a number, string, or bool. For now, I've created [constructors](https://github.com/DataDog/dd-trace-go/blob/5b4f5ceec47a4bb3c4fb9744764f1afb7073027e/internal/telemetry/message.go#LL127-L148C2) to initialize `Configuration` structs with `Value`'s of type string, number, or bool. 

The `Value` and `Name` fields are still public though to allow for testing key/values of configs e.g. [here](https://github.com/DataDog/dd-trace-go/blob/5b4f5ceec47a4bb3c4fb9744764f1afb7073027e/internal/telemetry/client_test.go#LL362-L374C31), and in the future in the tracer/profiler packages. Would this be ok, given that this package is currently for internal use?